rfd/0046-database-access-config.md

```diff
@@ -64,14 +66,269 @@ $ clusterawsadm bootstrap iam print-policy --document AWSIAMManagedPolicyControl
+of subcommands. This aligns with our strategy of placing commands related to a
+particular service under its subcommand namespace. It will provide subcommands
+for generating configuration files and bootstrapping database agents’ necessary
+configurations, for example, setting up IAM permissions for agents to proxy
+AWS-hosted databases.
+
+### Create config subcommand
+
+Similar to `teleport configure` but focused on configurations for Database
+agents. This command will generate a configuration file. It can also provide
+some examples of configuration, for instance, self-hosted databases setup.
+
+There will be flags to enable features like Aurora/RDS auto-discovery and
+dynamic registration. Each sample will show the required options with some
+explanation on how to fill them and documentation links.
+
+It will also provide flags to configure a specific database (similar to the
+flags present on the `teleport db start` command). When executed with these
+flags, the samples will not be included, only the static configuration of a
+single database.
+
+```bash
+teleport db configure create [--auth-server] [--token=] [--version=] [--enable-rds-auto-discovery] [--enable-dynamic-registration] [--output=] [--ca-pin] [--name] [--protocol] [--uri]
+```
+
+| Flag                            | Description |
+| ------------------------------- | ----------- |
+| `--auth-server`                 | Address of the auth server. |
+| `--token`                       | Invitation token to register with an auth server |
+| `--version`                     | Teleport configuration version. |
+| `--enable-rds-auto-discovery`   | Enables Aurora/RDS auto-discovery. |
+| `--enable-dynamic-registration` | Enables Dynamic registration. |
+| `--output`                      | Write to stdout with -o=stdout, default config file with -o=file or custom path with -o=file:///path |
+| `--ca-pin`                      | CA pin to validate the auth server (can be repeated for multiple pins). |
+| `--name`                        | Name of the proxied database. |
+| `--protocol`                    | Proxied database protocol. Supported are: [postgres mysql mongodb cockroachdb]. |
+| `--uri`                         | Address the proxied database is reachable at. |
+
+* When configuring a single database, `name`, `protocol` and `uri` are required;
+
+**Examples:**
+
+None of the flags are required. Running the command without flags should output
+a configuration with samples.
+
+```bash
+# will generate a configuration file for the database agent.
+$ teleport db configure create
+
+#
+# Teleport database agent configuration file.
+# Configuration reference: https://goteleport.com/docs/database-access/reference/configuration/
+#
+version: v2
+teleport:
+  nodename: localhost
+  data_dir: /var/lib/teleport
+  auth_token: /tmp/token
+  auth_servers:
+  - 127.0.0.1:3025
+db_service:
+  enabled: "yes"
+  # Lists statically registered databases proxied by this agent.
```

Comment:
I would include sample auto-discovery and dynamic registration sections as well. Probably even make them uncommented with some reasonable defaults, for example:

```yaml
  resources:
  - labels:
      "*": "*"

  aws:
  - types: ["rds"]
    regions: ["us-east-1", "us-west-1"]
    tags:
      "*": "*"
```

Reply:
I agree to add the samples, but I'm not sure about adding them uncommented. Users might infer that those configurations are required (like the global configuration).

Also, I think that the "shortest path" to run a database agent should be executed without any additional steps (like setting up AWS roles, etc.).

```bash
# Generate the configuration (without any extra option, except the file path)
$ teleport db configure create --output file:///teleport.yaml

# And then, just run it without modifying the configuration.
$ teleport start -c teleport.yaml
... successfully starts Teleport ...
```

And it might not be possible if we let those configurations be uncommented.

Reply:
Yeah I agree that `teleport db configure create` should just create a working config by default. Following up on my comment above, let's always include dynamic registration (it is our preferred method of registring dbs now anyway).

I'm also thinking it might be nice to auto-detect that command runs on an AWS instance and auto-enable RDS discover as well but that might be too smart. We can leave this out of scope for now.